### PR TITLE
Weather app UI Work

### PR DESCRIPTION
**Pull request by ghulamhafeez**
## Weather App Work

- Improved Weather App UI
- Added routes & Store 
- Created services with rtk query   
- 
## Weather Gif
-
weth

### CODE REVIEW

**Comment by Monfernape** [comment]
@ghulamhafeez build doesn't show the right UI.
https://deploy-preview-2--zingy-kitsune-6eb6a7.netlify.app/

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "formik": "^2.2.9",
+    "google-map-react": "^2.2.0",
```
Comment (Monfernape) [inline package.json]: Do we need this?

File: package.json [hunk 2]
```diff
@@ -11,11 +11,13 @@
...
     "history": "^5.3.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-redux": "^8.0.5",
     "react-router-dom": "^6.5.0",
-    "react-scripts": "5.0.1",
+    "react-scripts": "^2.1.3",
```
Comment (Monfernape) [inline package.json]: That's weird. Why did we have to downgrade `react-scripts`?

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
+  
+  ]);
   return (
     <div>
-
- <MainLayout/>
- </div>
+      <React.StrictMode>
+        <Provider store={store}>
+          <RouterProvider router={router}></RouterProvider>
```
Comment (Monfernape) [inline src/App.js:36]: Nice

File: src/services/WeatherService.js [hunk 4]
```diff
@@ -1,11 +1,33 @@
...
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 
-export const weatherApi = createApi({
-    reducerPath: 'weatherApi',
-    baseQuery: fetchBaseQuery({ baseUrl: '' }),
-    endpoints: (builder) => ({
-      getPokemonByName: builder.query<Pokemon, string>({
-        query: (name) => `pokemon/${name}`,
-      }),
+const apikey = "w74RorP4hf7wdUwIUlKNPhRxIa4fEVjW";
```
Comment (Monfernape) [inline src/services/WeatherService.js]: Let's keep it in constants, shall we?

File: src/services/WeatherService.js [hunk 5]
```diff
@@ -1,11 +1,33 @@
...
+
+export const WeatherApi = createApi({
+  reducerPath: "WeatherApi",
+  baseQuery: fetchBaseQuery({ baseUrl: "http://dataservice.accuweather.com" }),
```
Comment (Monfernape) [inline src/services/WeatherService.js]: Same for base url, move it to constants?

File: src/services/WeatherService.js [hunk 6]
```diff
@@ -1,11 +1,33 @@
...
-  })
\ No newline at end of file
+  }),
+});
+export const {
+  useGetTemperatureQuery,
+  useGetCityAutoCompleteQuery,
+  useGetDailyForecastQuery,
+  useGetHourlyForecastQuery
+} = WeatherApi;
```
Comment (Monfernape) [inline src/services/WeatherService.js:32]: Really liked the RTK setup

File: src/views/Weather-Components/CityCards.js [hunk 7]
```diff
@@ -8,36 +8,40 @@ import Button from "@mui/material/Button";
...
+    navigate(`/weather-detail/${id}`);
+  };
 
   return (
     <div>
-        <div className="cities-card">
+      <div className="cities-card">
         {POPULAR_CITIES.map((x) => {
           return (
             <Card
```
Comment (Monfernape) [inline src/views/Weather-Components/CityCards.js:21]: I think this could be a component.

File: src/views/Weather-Components/Constant.js [hunk 8]
```diff
@@ -14,10 +14,10 @@ export const Cities = [
 ];
 
 export const POPULAR_CITIES = [
-  { city: "Lahore", province: "Punjab" },
-  { city: "Multan", province: "Punjab" },
-  { city: "Kirachi", province: "Sindh" },
-  { city: "Quetta", province: "Balochistan" },
+  { city: "Lahore", Key: "260622", province: "Punjab" },
```
Comment (Monfernape) [inline src/views/Weather-Components/Constant.js]: Why isn't key lowercase?

File: src/views/Weather-Components/Header.js [hunk 9]
```diff
@@ -1,29 +1,78 @@
...
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import TextField from "@mui/material/TextField";
-import { Cities } from "./Constant";
 import Autocomplete from "@mui/material/Autocomplete";
+
+import { useGetCityAutoCompleteQuery } from "../../services/WeatherService";
 export const Header = () => {
+  const navigate = useNavigate();
+  const handleRedirect = (id) => {
```
Comment (Monfernape) [inline src/views/Weather-Components/Header.js]: Try to keep local functions below react hooks.

File: src/views/Weather-Components/Header.js [hunk 10]
```diff
@@ -1,29 +1,78 @@
...
+    navigate(`/weather-detail/${id}`);
+  };
+
+  const [autoCompleteValue, setAutoCompleteValue] = useState();
+
+  const { data } = useGetCityAutoCompleteQuery(autoCompleteValue);
+  console.log("d", data);
```
Comment (Monfernape) [inline src/views/Weather-Components/Header.js]: Unnecessary consoles.

File: src/views/Weather-Components/Header.js [hunk 11]
```diff
@@ -1,29 +1,78 @@
...
-                  ...params.InputProps,
-                  type: "search",
-                }}
-              />
-            )}
-          />
+      <div>
+        <div className="search-div">
+          <div className="search">
+            <Autocomplete
```
Comment (Monfernape) [inline src/views/Weather-Components/Header.js]: Let's make it a component, shall we?

File: src/views/Weather-Components/WeatherDetail.js [hunk 12]
```diff
@@ -1,44 +1,55 @@
...
+} from "../../services/WeatherService";
+import { useParams } from "react-router-dom";
+import { WeatherDetailCards } from "./WeatherDetailCards";
 export const WeatherDetail = () => {
+  const { id } = useParams();
+  const { data } = useGetDailyForecastQuery(id);
+
+  const handleRedirect = () => {
+    const { data } = useGetHourlyForecastQuery(id);
+    console.log("houlryData", data);
```
Comment (Monfernape) [inline src/views/Weather-Components/WeatherDetail.js]: Oops.

**Review by Monfernape — CHANGES_REQUESTED** [review]

**Comment by Monfernape** [comment]
Also, your designs are not mobile responsive.

File: package.json (hunk 1 above)
Comment (ghulamhafeez) [reply package.json]: No i will remove

File: src/App.js (hunk 3 above)
Comment (ghulamhafeez) [reply src/App.js:36]: Thanks

File: src/services/WeatherService.js (hunk 4 above)
Comment (ghulamhafeez) [reply src/services/WeatherService.js]: ok

File: src/services/WeatherService.js (hunk 5 above)
Comment (ghulamhafeez) [reply src/services/WeatherService.js]: ok

File: src/views/Weather-Components/Header.js (hunk 9 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/Header.js]: ok

File: src/views/Weather-Components/Constant.js (hunk 8 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/Constant.js]: sorry i will fix it

File: src/views/Weather-Components/Header.js (hunk 11 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/Header.js]: ok

File: src/views/Weather-Components/WeatherDetail.js (hunk 12 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/WeatherDetail.js]: i will remove it

File: src/views/Weather-Components/Header.js (hunk 10 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/Header.js]: ok

File: src/services/WeatherService.js (hunk 6 above)
Comment (ghulamhafeez) [reply src/services/WeatherService.js:32]: Thanks

File: src/views/Weather-Components/CityCards.js (hunk 7 above)
Comment (ghulamhafeez) [reply src/views/Weather-Components/CityCards.js:21]: ok

File: package.json (hunk 2 above)
Comment (ghulamhafeez) [reply package.json]: i will fix

**Review by Monfernape — APPROVED** [review]

**Pull request merged**